### ISSUE TITLE
🐛  TS4023 from removing reported unused export if exported type is referenced implicitly in another package 

### ISSUE THREAD

**Issue by aaayane** [webpro-nl/knip#808]
### Prerequisites

- [X] I'm using the latest version
- [X] I've read the relevant [documentation](https://knip.dev)
- [X] I've searched for [existing issues](https://github.com/webpro-nl/knip/issues?q=is%3Aissue)
- [X] I've checked the [list of known issues](https://knip.dev/reference/known-issues)
- [X] I've read the [issue reproduction guide](https://knip.dev/guides/issue-reproduction)

### Reproduction url

https://stackblitz.com/edit/github-a4rrlb-3x1iep?file=README.md

### Reproduction access

- [X] I've made sure the reproduction is publicly accessible

### Description of the issue

Hi, thanks for your work on this project. I think I have a similar problem which doesn't seem to be solved in the new version.(knip 5.33.2).
from [https://github.com/webpro-nl/knip/issues/801](url)

When exporting a function, the reference type will still prompt that it is not used

Reproduction url
[https://stackblitz.com/edit/github-a4rrlb-3x1iep?file=README.md](https://github.com/webpro-nl/knip/issues/url)

Description of the issue
When exporting a function, the reference type will still prompt that it is not used
TS4023: Exported variable 'sharedFunction' has or is using name 'Error1' from external module "/home/projects/github-a4rrlb/packages/shared/src/file2" but cannot be named.

See the readme of the linked reproduction for exactly how this can happen

**Comment by webpro** [comment]
Thanks for the report, @aaayane! Could you please try with this one:

```
npm i -D https://pkg.pr.new/knip@1c04163
```

Happy to know if this works better for you!

**Comment by aaayane** [comment]
> https://github.com/webpro-nl/knip/issues/801

Looks like this has been resolved, thanks for your work!

**Comment by aaayane** [comment]
> > #801
> 
> Looks like this has been resolved, thanks for your work!

@webpro By the way, I may have found another problem

### Reproduction url
[https://stackblitz.com/edit/github-a4rrlb-7fzqsb?file=README.md,packages%2Fshared%2Fsrc%2Ffile2.ts](url)

### Description of the issue
When I export destructuring assignment, using fix will mistakenly delete the variable being used

When I wrote the example, I found that using npm only prompts unused exports, but using pnpm will delete the definition of a.

I don’t know what the difference is between the two. It seems that there is an error when using pnpm。

See the readme of the linked reproduction for exactly how this can happen

**Comment by webpro** [comment]
Thanks @aaayane! Very happy with your feedback. Things should be better now here:

```
npm i -D https://pkg.pr.new/knip@307ef8d
```

Ideally, the `export` keyword is removed when there are no exported elements left, but that'll come later. At least the referenced exported element is no longer removed.

**Comment by aaayane** [comment]
> Thanks @aaayane! Very happy with your feedback. Things should be better now here:
> 
> ```
> npm i -D https://pkg.pr.new/knip@307ef8d
> ```
> 
> Ideally, the `export` keyword is removed when there are no exported elements left, but that'll come later. At least the referenced exported element is no longer removed.

it's worked, , thanks for your work, this project is great and solved my problem

**Comment by webpro** [comment]
Nice! The implementation is something I need to clean up, and removing the `export` keyword for such empty bound declarations is on my list to do. But I'm very glad we got this sorted for now.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.33.3. See [Release 5.33.3](https://github.com/webpro-nl/knip/releases/tag/5.33.3) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._
